src/org/labkey/test/pages/query/EditMetadataPage.java

```diff
@@ -0,0 +1,88 @@
+package org.labkey.test.pages.query;
+
+import org.labkey.test.BaseWebDriverTest;
+import org.labkey.test.Locator;
+import org.labkey.test.WebDriverWrapper;
+import org.labkey.test.components.bootstrap.ModalDialog;
+import org.labkey.test.components.domain.BaseDomainDesigner;
+import org.labkey.test.components.domain.DomainFormPanel;
+import org.labkey.test.components.react.ReactSelect;
+import org.labkey.test.util.DataRegionTable;
+import org.openqa.selenium.WebDriver;
+import org.openqa.selenium.WebElement;
+
+import static org.labkey.test.WebDriverWrapper.WAIT_FOR_JAVASCRIPT;
+
+public class EditMetadataPage extends BaseDomainDesigner<EditMetadataPage.ElementCache>
+{
+    public EditMetadataPage(WebDriver driver)
+    {
+        super(driver);
+    }
+
+
+    public DataRegionTable clickViewData(BaseWebDriverTest test, String schemaName, String queryName)
+    {
+        return test.viewQueryData(schemaName, queryName);
+    }
+
+    public void clickEditSourcePage()
+    {
+        elementCache().editSourceBtn.click();
+    }
+
+    public EditMetadataPage clickSave()
+    {
+        elementCache().saveBtn.click();
+        return this;
+    }
+
+    public void clickResetToDefault()
+    {
+        WebDriverWrapper.waitFor(elementCache().resetToDefaultBtn::isEnabled,
+                "The alias button did not become enabled as expected", 2000);
+        elementCache().resetToDefaultBtn.click();
+        new ModalDialog(new ModalDialog.ModalDialogFinder(getDriver()).withTitle("Confirm Reset")).dismiss("Reset");
+    }
```

Comment:
In this case maybe the function should just be called `resetToDefault`, because it does more than just click the button it also responds to the dialog. 

Alternatively you could leave it named as is and return the dialog and let the test deal with it.

Generally when I see a "click" method I'm expecting some singular action that results in something like a page navigation, form submit or returning some prompt I need to interact with. This might be something to discuss during our next coding standards discussion.

Reply:
Agree